lib/rest/routes/apps.js

```diff
@@ -8,7 +9,15 @@ module.exports = function () {
   router.post('/', function (req, res, next) {
     const userId = req.body.userId;
     delete req.body.userId;
-    applicationsSrv.insert(req.body, userId)
+
+    findUser(userId)
```

Comment:
this logic should be in the service `userSrv.get(userIdOrUsername)`

Reply:
You're right, it might make sense. I'll have a look to the code and see if I can move it.

Reply:
@DrMegavolt 

I've checked and you're totally right. These things should go in the services.

Would it be ok to create a new issue and work on this as a separate pull request? This one is already big and in order to move these function in the services there are some errands to do (such as function renaming) and I do not want to mess up the tests here.

If you agree, I'll create a new issue.

Reply:
I'll be reviewing it all together. It is hard for me to keep in mind all my comments between PRs. If it is easier for you, np, just make it on top of this branch so it will look as mega massive PR. 


btw, what you will have is:
PR1 
apps.js -1 line +1 line (findUser)
usage of findUser everywhere  +N lines

PR2
rm findUser usage -N lines
revert app.js  -1 +1
copy logic for find user from rest interface to the service -X +X lines 

so for review it will be 
+ (1+N+X+1) -(1+N+X+1) 

If you do it in the same PR 
it will be only +X -X because you just move code

Reply:
I like the math you did!
All right, I'll just push one more commit to move these parts 👍